llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -111,13 +111,13 @@ static std::optional<uint64_t> getObjectSize(const Value *V,
   Opts.RoundToAlign = RoundToAlign;
   Opts.NullIsUnknownSize = NullIsValidLoc;
   if (getObjectSize(V, Size, DL, &TLI, Opts))
-    return Size;
+    return TypeSize::getFixed(Size);
   return std::nullopt;
 }
 
 /// Returns true if we can prove that the object specified by V is smaller than
 /// Size.
-static bool isObjectSmallerThan(const Value *V, uint64_t Size,
+static bool isObjectSmallerThan(const Value *V, LocationSize Size,
```

Comment:
I think if you make this accept `TypeSize` instead, and then...

Reply:
ok